pkg/worker/worker_test.go

```diff
@@ -0,0 +1,174 @@
+package worker
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	googleUUID "github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+type MemoryStorage struct {
+	Data map[googleUUID.UUID]Job
+	sync.Mutex
+	MemoryErrors
+}
+
+type MemoryErrors struct {
+	StoreError    error
+	RetrieveError error
+}
+
+func (m *MemoryStorage) Store(job Job) error {
+	m.Data[job.UUID] = job
```

Comment:
Sharing the underlying slice in `job.Logs` could lead to data races.
Either do a deep copy of `job.Logs` when storing or use some memory-safe type to keep logs instead

Reply:
I applied the changes. Is this acceptable way of deep copying? I know it happens after the job is copied along with underlying slice, but both lines are surrounded by Mutex.Lock, so no other data should access the slice while it's being copied. Do I understand it correctly?

Reply:
you need to deep-copy on `Retrieve` too.
otherwise, I can:
```
j, err := store.Retrieve(uuid)
// handle err
j.Logs[0] = "something else" // this races with anyone reading Logs concurrently
```

but, see the suggestion from @russjones above about reading the job output.
it may change how the data store deals with output too.